python/openassetio/log.py

```diff
@@ -93,14 +93,20 @@ def getSeverity(self):
     # LoggerInterface methods
 
     def log(self, message, severity):
-
+        """
+        Log only if severity is less than or equal to maximum allowed
```

Comment:
[fix] Here (and progress, but that's going away...). The concept of min/max for severity is always a head-mess. I think we have been referring to the "logical level" ie "critical is high, debug is low", rather than the enum value. The docs at L33 suggest messages below the specified severity are dropped, so I think this needs to be "Log only if severity is greater or equal to the loggers configured severity level" (or similar).

Reply:
If we want to change the enum around that's totally fine too!

Reply:
Updated the docstrings in e0862b82.  

I also added tests for the severity names in c536ceb4, in preparation for reversing the severity enum order.  However, it occurred to me that we're more likely to add lower logical severities than higher (e.g. a `kTrace` severity), so having lower logical severities with higher enum values allows us to easily add more.  

It is a bit mind-bending though. I'm happy to reverse the enum values if you think its worthwhile.

Reply:
Can we [plan] an Issue for re-ordering? And we can pick it up once this slice is done. Good idea on the tests 👍

Reply:
Re. leaving some room, a good idea, but kinda breaks the `kSeverityNames` lookup, so we'd have to rethink that.

Reply:
Added https://github.com/OpenAssetIO/OpenAssetIO/issues/516 to re-order severity enums.